add workflow that builds bliss on different platforms for release or debug

matbesancon opened the pull request:
This will probably break for now. I took the structure from https://github.com/ERGO-Code/HiGHS/blob/master/.github/workflows/build-windows.yml

matbesancon committed:
add unix workflow

fschloesser commented:
looks good to me so far, i think we could even try to add `windows-latest` for the os.

matbesancon commented:
ping @gasse, can you activate / allow workflows for this PR?

gasse commented:
Hi @matbesancon , I have no idea how to activate / allow workflows :P Do you want me to merge this pull request? Maybe @AntoinePrv knows better what to do?

matbesancon commented:
it might require allowing actions to run, either in this PR if there is the option, or at the repo level yes

AntoinePrv commented:
> The workflow is not valid. .github/workflows/ci.yml (Line: 15, Col: 9): A sequence was not expected

These Yaml configs are really a nightmare to debug...

svigerske committed:
merge debug and release into same job

- adding buildtype to matrix

svigerske committed:
arch has no effect; select by runner label; add windows

- add runs on ubuntu-24.04-arm, intel-macos, windows-latest, and
  windows-arm runners
